grin/src/adapters.rs

```diff
@@ -376,20 +375,33 @@ impl NetToChainAdapter {
 
 	// pushing the new block through the chain pipeline
 	// remembering to reset the head if we have a bad block
-	fn process_block(&self, b: core::Block) -> bool {
-		let bhash = b.hash();
+    fn process_block(&self, b: core::Block, addr: SocketAddr) -> bool {
+        let prev_hash = b.header.previous;
+        let bhash = b.hash();
 		let chain = w(&self.chain);
-		let res = chain.process_block(b, self.chain_opts());
-		if let Err(ref e) = res {
-			debug!(LOGGER, "Block {} refused by chain: {:?}", bhash, e);
-			if e.is_bad_data() {
-				debug!(LOGGER, "adapter: process_block: {} is a bad block, resetting head", bhash);
-				let _ = chain.reset_head();
-				return false;
-			}
-		};
-		true
-	}
+	    match chain.process_block(b, self.chain_opts()) {
+                Ok(_) => true,
+                Err(e) => {
+			        debug!(LOGGER, "Block {} refused by chain: {:?}", bhash, e);
+                    match e {
```

Comment:
I would merge that match into the one above, the debug statement can just be moved into the `true` case as the other branches are already covered.

Reply:
Tried to preserve the original log message, but you are right, we have specific ones in each case